### PR TITLE
fix(asset): cannot create asset if cwip disabled and account not set

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: [ASANA](https://app.asana.com/0/home/1192407894347497/1198992822363359)
Changes port from [fix(asset): cannot create asset if cwip disabled and account not set](https://github.com/frappe/erpnext/pull/23580)

### CODE REVIEW

File: erpnext/assets/doctype/asset/asset.py [hunk 1]
```diff
@@ -459,50 +459,63 @@ def get_default_finance_book_idx(self):
...
+		asset_bought_with_invoice = (purchase_document == self.purchase_invoice)
+		fixed_asset_account = self.get_fixed_asset_account()
+		
 		cwip_enabled = is_cwip_accounting_enabled(self.asset_category)
-		# check if expense already has been booked in case of cwip was enabled after purchasing asset
-		expense_booked = False
-		cwip_booked = False
+		cwip_account = self.get_cwip_account(cwip_enabled=cwip_enabled)
 
+		query = """SELECT name FROM `tabGL Entry` WHERE voucher_no = %s and account = %s"""
```
Comment (neilLasrado) [inline erpnext/assets/doctype/asset/asset.py]: @vishdha cant we do get doc with Filters instead of this query ?

Comment (hrwX) [reply erpnext/assets/doctype/asset/asset.py]: @vishdha could you make the changes

**Comment by neilLasrado** [comment]
@vishdha please rework this on priority

File: erpnext/assets/doctype/asset_category/asset_category.py [hunk 2]
```diff
@@ -58,6 +59,21 @@ def validate_account_types(self):
...
+			for d in self.accounts:
+				if (not d.capital_work_in_progress_account and 
+					not frappe.db.get_value("Company", d.company_name, "capital_work_in_progress_account")):
+					missing_cwip_accounts_for_company.append(get_link_to_form("Company", d.company_name))
+
+			if missing_cwip_accounts_for_company:
+				msg = _("""To enable Capital Work in Progress Accounting, """)
+				msg += _("""you must select Capital Work in Progress Account in accounts table""")
+				msg += "<br><br>"
+				msg += _("You can also set default CWIP account in Company {}").format(", ".join(missing_cwip_accounts_for_company))
```
Comment (hrwX) [inline erpnext/assets/doctype/asset_category/asset_category.py]: Can we merge the msg ?

Comment (vishdha) [reply erpnext/assets/doctype/asset_category/asset_category.py]: yup. I modified a little bit.

**Review by hrwX — APPROVED** [review]

**Pull request merged**